corpus/data_augmentation_audiomentations.py

```diff
@@ -0,0 +1,162 @@
+__all__ = ["AddImpulseResponseJob", "MixNoiseJob"]
+
+import os
+import audiomentations
+import shutil
+from scipy.io import wavfile
+
+from sisyphus import *
+from recipe.i6_core.lib import corpus
+
+
+class AddImpulseResponseJob(Job):
+    """
+    Add impulse responses to each recording in the corpus with a given probability to be affected.
+    This is done by convolving the audio with an impulse response to simulate other acoustic conditions.
+    Supports .wav files
+    """
+
+    def __init__(
+        self, corpus_file, new_corpus_name, ir_path, augment_prob, time_rqmt=12
+    ):
+        """
+
+        :param Path corpus_file: Bliss corpus with wav files
+        :param str new_corpus_name: name of the new corpus
+        :param ir_path: path to the directory containing impulse responses
+        :param augment_prob: probability of each audio to be augmented by impulse response convolution
+        :param time_rqmt: time requirement for the Sisypus job
+        """
+        self.corpus_file = corpus_file
+        self.new_corpus_name = new_corpus_name
+        self.impulse_responses_dir = ir_path
+
+        self.out_audio_folder = self.output_path("out_audio/", directory=True)
+        self.out_corpus = self.output_path("ir_convoluted.corpus.xml.gz")
+
+        self.rqmt = {"time": time_rqmt, "cpu": 2}
+
+        self.add_impulse_response = audiomentations.AddImpulseResponse(
+            ir_path=ir_path, p=augment_prob
+        )
+
+    def tasks(self):
+        yield Task("run", rqmt=self.rqmt)
+
+    def run(self):
+        job_id = os.path.basename(self.job_id())
+        orig_corpus = corpus.Corpus()
+        new_corpus = corpus.Corpus()
+
+        orig_corpus.load(tk.uncached_path(self.corpus_file))
```

Comment:
`self.corpus_file.get_path()`, we want to get rid of string casts, see https://github.com/rwth-i6/i6_core/issues/32